### PR TITLE
Remove duplicated definitions for .min() and .max() functions

### PR DESCRIPTION
**Pull request by ljmf00**
Signed-off-by: Luís Ferreira <contact@lsferreira.net>

### CODE REVIEW

File: src/core/internal/util/math.d [hunk 1]
```diff
@@ -0,0 +1,53 @@
+// Written in the D programming language
+
+/**
+ * Internal math utilities.
+ *
+ * Copyright: Luís Ferreira 2021
```
Comment (ljmf00) [inline src/core/internal/util/math.d:6]: I'm not really sure if this should be my or D Programming Language Foundation copyrights. Can someone please clarify it to me?

Comment (wilzbach) [reply src/core/internal/util/math.d:6]: Should be DLF. Old code only has it for historical reasons.

Comment (ljmf00) [reply src/core/internal/util/math.d]: Force pushed :)

File: src/core/internal/util/math.d [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+ *   b = second value to select the minimum from
+ * Returns: The minimum of the passed-in values.
+ */
+T min(T)(T a, T b) pure nothrow @nogc @safe
+{
+    return b < a ? b : a;
+}
+
+///
+@safe pure
```
Comment (RazvanN7) [inline src/core/internal/util/math.d:38]: `@nogc nothrow`

File: src/core/internal/util/math.d [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+unittest
+{
+    assert(max(1,3) == 3);
+    assert(max(3,1) == 3);
+    assert(max(1,1) == 1);
+}
+
+///
+@safe pure
```
Comment (RazvanN7) [inline src/core/internal/util/math.d:47]: ditto

**Review by RazvanN7 — COMMENTED** [review]
Creating a new module just for those functions seems like an overkill to me, however, the duplication is also annoying. @schveiguy you have more experience with druntime, what do you think about this?

**Comment by ljmf00** [comment]
> Creating a new module just for those functions seems like an overkill to me, however, the duplication is also annoying. @schveiguy you have more experience with druntime, what do you think about this?

Yes, I understand, although I thought it would be fine because it will:
- remove some duplication and possibly other future math duplications out there
- not introduce anything new to the public API nor breaking changes.

**Comment by maxhaton** [comment]
> Creating a new module just for those functions seems like an overkill to me, however, the duplication is also annoying. @schveiguy you have more experience with druntime, what do you think about this?

Maybe the name could be improved (something more general to avoid future modules?) but it seems likely that this situation will happen again in the future so both setting the standard early of avoiding duplication and avoiding encouraging putting generic stuff in one file is a good thing.

File: src/core/internal/util/math.d (hunk 2 above)
Comment (ljmf00) [reply src/core/internal/util/math.d]: Done

File: src/core/internal/util/math.d (hunk 3 above)
Comment (ljmf00) [reply src/core/internal/util/math.d]: Done

**Review by RazvanN7 — APPROVED** [review]

File: src/core/internal/util/math.d [hunk 4]
```diff
@@ -0,0 +1,53 @@
+// Written in the D programming language
```
Comment (dkorpel) [inline src/core/internal/util/math.d:1]: What's the point of this line?

Comment (ljmf00) [reply src/core/internal/util/math.d:1]: I'm not really sure honestly, but I just followed the criteria around some files in `druntime`/`phobos`.

e.g.:
https://github.com/dlang/druntime/blob/f94e816c1dd229cbb133085a6fc41d1bd6d45594/src/core/math.d#L1

**Pull request merged**